src/components/Languages/Json/ColorPicker/Data.ts

```diff
@@ -0,0 +1,37 @@
+import { markRaw } from 'vue'
+import { App } from '/@/App'
+import { Signal } from '/@/components/Common/Event/Signal'
+
+export class ColorData extends Signal<void> {
+	protected _data?: any
+
+	async loadColorData() {
+		const app = await App.getApp()
+
+		this._data = markRaw(
+			await app.dataLoader.readJSON(
+				`data/packages/minecraftBedrock/location/validColor.json`
+			)
+		)
+
+		this.dispatch()
+	}
+
+	async getDataForCurrentTab() {
+		if (!this._data)
```

Comment:
Throwing an error here is a bit unnecessary as we can simply `await this.fired` instead